modules/appeals_api/spec/requests/v1/notice_of_disagreements/evidence_submissions_controller_spec.rb

```diff
@@ -5,66 +5,140 @@
-        allow(s3_bucket).to receive(:object).and_return(s3_object)
-        allow(s3_object).to receive(:presigned_url).and_return(+'http://some.fakesite.com/path/uuid')
-        yield
+    context 'when corresponding notice of disagreement record not found' do
+      it 'returns an error' do
+        with_settings(Settings.modules_appeals_api.evidence_submissions.location,
+                      prefix: 'https://fake.s3.url/foo/',
+                      replacement: 'https://api.vets.gov/proxy/') do
+          s3_client = instance_double(Aws::S3::Resource)
+          allow(Aws::S3::Resource).to receive(:new).and_return(s3_client)
+          s3_bucket = instance_double(Aws::S3::Bucket)
+          s3_object = instance_double(Aws::S3::Object)
+          allow(s3_client).to receive(:bucket).and_return(s3_bucket)
+          allow(s3_bucket).to receive(:object).and_return(s3_object)
+          allow(s3_object).to receive(:presigned_url).and_return(+'https://fake.s3.url/foo/uuid')
+          post(path, params: { nod_id: 1979, headers: headers })
+
+          expect(response.status).to eq 404
+          expect(response.body).to include 'not found'
+        end
       end
     end
-    context "when 'nod_id' is included" do
-      it "returns submission attributes including 'appealId'" do
-        with_s3_settings do
+
+    context 'when corresponding notice of disagreement record found' do
+      it "returns an error if nod 'boardReviewOption' is not 'evidence_submission'" do
+        with_settings(Settings.modules_appeals_api.evidence_submissions.location,
+                      prefix: 'https://fake.s3.url/foo/',
+                      replacement: 'https://api.vets.gov/proxy/') do
+          s3_client = instance_double(Aws::S3::Resource)
+          allow(Aws::S3::Resource).to receive(:new).and_return(s3_client)
+          s3_bucket = instance_double(Aws::S3::Bucket)
+          s3_object = instance_double(Aws::S3::Object)
+          allow(s3_client).to receive(:bucket).and_return(s3_bucket)
+          allow(s3_bucket).to receive(:object).and_return(s3_object)
+          allow(s3_object).to receive(:presigned_url).and_return(+'https://fake.s3.url/foo/uuid')
           post(path, params: { nod_id: notice_of_disagreement.id, headers: headers })
+
+          expect(response.status).to eq 422
+          expect(response.body).to include "'boardReviewOption' must be 'evidence_submission'"
         end
+      end
 
-        data = JSON.parse(response.body)['data']
+      context "when nod record 'auth_headers' are present" do
+        it "returns an error if request 'headers['X-VA-SSN'] and NOD record SSNs do not match" do
+          with_settings(Settings.modules_appeals_api.evidence_submissions.location,
+                        prefix: 'https://fake.s3.url/foo/',
+                        replacement: 'https://api.vets.gov/proxy/') do
+            s3_client = instance_double(Aws::S3::Resource)
```

Comment:
Can we extract this whole s3 doubling somewhere into a shared spot? Even if it's a method in this spec file, that's fine. So much boilerplate that's really tangential to what we're actually trying to test.

Reply:
I had it pulled out, but it appears to be causing a build bug, or so we thought. That was a whole thing while you were out.